numba/np/math/cmathimpl.py

```diff
@@ -0,0 +1,542 @@
+"""
+Implement the cmath module functions.
+"""
+
+
+import cmath
+import math
+
+from numba.core.imputils import impl_ret_untracked
+from numba.core import types
+from numba.core.typing import signature
+from numba.cpython import mathimpl
+from numba.core.extending import overload
+
+# registry = Registry('cmathimpl')
+# lower = registry.lower
+
+
+def is_nan(builder, z):
+    return builder.fcmp_unordered('uno', z.real, z.imag)
+
+def is_inf(builder, z):
+    return builder.or_(mathimpl.is_inf(builder, z.real),
+                       mathimpl.is_inf(builder, z.imag))
+
+def is_finite(builder, z):
+    return builder.and_(mathimpl.is_finite(builder, z.real),
+                        mathimpl.is_finite(builder, z.imag))
+
+
+# @lower(cmath.isnan, types.Complex)
+def isnan_float_impl(context, builder, sig, args):
+    [typ] = sig.args
+    [value] = args
+    z = context.make_complex(builder, typ, value=value)
+    res = is_nan(builder, z)
+    return impl_ret_untracked(context, builder, sig.return_type, res)
+
+# @lower(cmath.isinf, types.Complex)
+def isinf_float_impl(context, builder, sig, args):
+    [typ] = sig.args
+    [value] = args
+    z = context.make_complex(builder, typ, value=value)
+    res = is_inf(builder, z)
+    return impl_ret_untracked(context, builder, sig.return_type, res)
+
+
+# @lower(cmath.isfinite, types.Complex)
+def isfinite_float_impl(context, builder, sig, args):
+    [typ] = sig.args
+    [value] = args
+    z = context.make_complex(builder, typ, value=value)
+    res = is_finite(builder, z)
+    return impl_ret_untracked(context, builder, sig.return_type, res)
+
+
+@overload(cmath.rect)
```

Comment:
```suggestion
# @overload(cmath.rect)
```
it's duplicating registration with `cpython` https://github.com/numba/numba/blob/399bbf6c6152b15f70c3110f32d83959c94b997e/numba/cpython/cmathimpl.py#L57